### ISSUE TITLE
enhancement: have the vnc screen's canvas be the native resolution of the guest

### ISSUE THREAD

**Issue by breadtf** [cockpit-project/cockpit-machines#1390]
Having the canvas resolution be greater than that of the guest leads to the guest looking blurry. Removing the scaling (or at least adding the option to toggle it) makes the guest much sharper and clearer. It also allows the whole screen to be viewed on smaller devices.

Here is a before and after of the scaling off (using a css hack)

Before:
![Screenshot_20240116_201154](https://github.com/cockpit-project/cockpit-machines/assets/103989916/557c3408-43ab-46ef-a8cb-b0be124b54af)

After:
![Screenshot_20240116_200941](https://github.com/cockpit-project/cockpit-machines/assets/103989916/e7a2e764-7165-4794-a1f3-b52e0ef2ff5c)

The issue doesn't come across well on screenshots, but zooming in you can clearly see the pixels are way sharper.

Before:
![Screenshot_20240116_201333](https://github.com/cockpit-project/cockpit-machines/assets/103989916/880ed09f-d69d-46d8-abe7-f8dc96082511)

And after:
![Screenshot_20240116_201415](https://github.com/cockpit-project/cockpit-machines/assets/103989916/db08aa59-89a7-4f50-a482-64a41cb64885)

**Comment by garrett** [comment]
I agree! Optimizing for native resolution would be a good idea, especially when the host's resolution is less than the available space.

**Comment by BreadTeleporter** [comment]
Had a look at react-console's source (which the vnc viewer uses) and saw this
![image](https://github.com/cockpit-project/cockpit-machines/assets/103989916/b9a72532-ee5e-4895-9b4a-48901d2b2b62)
This looks like what's causing this annoying scaling, and, sure enough, there it was 
(in `src/components/vm/consoles/vnc.jsx`)
![image](https://github.com/cockpit-project/cockpit-machines/assets/103989916/91d5cc22-4be1-41e9-9c48-7f732e1426a6)
A quick and dirty fix could just be to remove this line, as by default it's set to false. I'll test this solution later today and I'll let you all know if it worked.

**Comment by BreadTeleporter** [comment]
Oh yeah, you're right. Completely forgot about that (Probably because I haven't used it)

That would probably involve creating an issue on react-console's github, as I doubt it would be possible here.
I'm not going to, though, because I'm not up for a wild goose chase trying to fix some minor issue. I'll just have to live with it for now and maybe make an issue later if it becomes more of a problem in future.

**Comment by garrett** [comment]
I think this is a valuable point to consider, and if we leave the issue open, it could serve as a reminder when we're (eventually) working on implementing the redesign of the in-page VNC viewer.

And, besides, react-console was originally from Cockpit Machines, so we can (and should... and even plan to) change it.

Thanks again for pointing out this issue!

**Comment by mvollmer** [comment]
Should be fixed since #2015.
